server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -0,0 +1,385 @@
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.NumericUtils;
+import org.opensearch.common.lucene.BytesRefs;
+import org.opensearch.common.lucene.Lucene;
+import org.opensearch.index.compositeindex.datacube.startree.index.StarTreeValues;
+import org.opensearch.index.compositeindex.datacube.startree.utils.iterator.SortedSetStarTreeValuesIterator;
+import org.opensearch.index.compositeindex.datacube.startree.utils.iterator.StarTreeValuesIterator;
+import org.opensearch.index.mapper.KeywordFieldMapper.KeywordFieldType;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.mapper.NumberFieldMapper;
+import org.opensearch.index.mapper.NumberFieldMapper.NumberFieldType;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.ExactMatchDimFilter;
+import org.opensearch.search.startree.filter.MatchNoneFilter;
+import org.opensearch.search.startree.filter.RangeMatchDimFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.BYTE;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.DOUBLE;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.FLOAT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.HALF_FLOAT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.INTEGER;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.LONG;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.SHORT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.hasDecimalPart;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.signum;
+
+public interface DimensionFilterMapper {
+    DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues);
+
+    DimensionFilter getRangeMatchFilter(
+        MappedFieldType mappedFieldType,
+        Object rawLow,
+        Object rawHigh,
+        boolean includeLow,
+        boolean includeHigh
+    );
+
+    Optional<Long> getMatchingOrdinal(
+        String dimensionName,
+        Object value,
+        StarTreeValues starTreeValues,
+        DimensionFilter.MatchType matchType
+    );
+
+    class Factory {
+
+        private static final Map<NumberFieldMapper.NumberType, DimensionFilterMapper> NUMERIC_SINGLETON_MAPPINGS = Map.of(
+            BYTE,
+            new IntegerFieldMapperNumeric(),
+            SHORT,
+            new IntegerFieldMapperNumeric(),
+            INTEGER,
+            new IntegerFieldMapperNumeric(),
+            LONG,
+            new SignedLongFieldMapperNumeric(),
+            HALF_FLOAT,
+            new HalfFloatFieldMapperNumeric(),
+            FLOAT,
+            new FloatFieldMapperNumeric(),
+            DOUBLE,
+            new DoubleFieldMapperNumeric()
```

Comment:
Lets add todo for unsigned long.

Also how does scaled float get handled here ? scaled float is a supported dimension which is present as part of modules

Reply:
Ack will add a TODO for unsigned long.

Scaled Float along with other mapped field types ( that can be created by plugins by extending MapperPlugin ) will have this problem since they are not present in OS Core. 

So, we will have to expose methods via the existing `MappedFieldType` which will overridden by the custom mapped types of plugin. 

Given this might be a significant change, I think we should pick it up separately rather than accommodating within this PR post 2.19